amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/Scheduler.java

```diff
@@ -612,6 +623,51 @@ void cleanupShardConsumers(Set<ShardInfo> assignedShards) {
         }
     }
 
+    /**
+     * Exceptions in the RxJava layer can fail silently unless an error handler is set to propagate these exceptions
+     * back to the KCL, as is done below. This method is internal to the class and has package access solely
+     * for testing.
+     *
+     * @param handler This method accepts a handler param solely for testing. All non-test calls to this method will
+     *                have a null input.
+     */
+    @VisibleForTesting
+    void registerErrorHandlerForUndeliverableAsyncTaskExceptions(@Nullable Consumer<Throwable> handler) {
+       if (handler == null) {
+           RxJavaPlugins.setErrorHandler(t -> {
+               ExecutorStateEvent executorStateEvent = new ExecutorStateEvent(executorService, leaseCoordinator);
+               RejectedTaskEvent rejectedTaskEvent = new RejectedTaskEvent(executorStateEvent, t);
+               rejectedTaskEvent.accept(diagnosticEventHandler);
+           });
+       } else {
+           RxJavaPlugins.setErrorHandler(t -> handler.accept(t));
+       }
+    }
+
+    private void logExecutorState() {
+        ExecutorStateEvent executorStateEvent = new ExecutorStateEvent(executorService, leaseCoordinator);
+        DiagnosticEventHandler debugHandler = new DiagnosticEventHandler() {
```

Comment:
we should move all this logic into DiagnosticEventLogger. That's the class responsible for logging details. DiagnosticEventLogger should be the one deciding when to log debug/info at what frequency.

Reply:
Great call. Moved all timing and level logic into DiagnosticEventLogger